### DISCUSSION TITLE
When I use the Prefetch hook to preload data, I find that there is no API network request when I access the link console directly from the browser. If I click the route to enter the sub-page of the console, there is no API network request when I click the route to jump to the rotor page. How can I solve this problem

### DISCUSSION THREAD

**Discussion by gh81997167** [quasarframework/quasar#13133] [CLI - SSR mode] [answered]
When I use the Prefetch hook to preload data, I find that there is no API network request when I access the link console directly from the browser. If I click the route to enter the sub-page of the console, there is no API network request when I click the route to jump to the rotor page. How can I solve this problem。

**The desired result**
Whether you visit the product details page directly from a browser or click on an item on the home page to get to the product details page, you can preload data and don't want to see API network requests on the Console.

**My router.js**
![image](https://user-images.githubusercontent.com/10526983/162553799-bd6d4749-d9f2-4cc2-9228-1c10ab62f113.png)

**Prefetch code**
![163B2ECB-A9EC-4faf-A9AE-1FB7F1EAB173](https://user-images.githubusercontent.com/10526983/162553978-3c053cd1-369e-4d67-acba-c2d9dd7c49cf.png)

**Console screenshot**
![image](https://user-images.githubusercontent.com/10526983/162553887-2763fd8a-02fc-4bd0-b8e0-bbe89055a16a.png)
![image](https://user-images.githubusercontent.com/10526983/162553898-7b200e41-a6eb-48e2-95db-0349d86cf2df.png)

**Comment by rstoenescu** [comment, marked as answer]
Hi,

When using SSR, the preFetch occurs on server and populates the page. Then, when client-side takes over, there is no need to fetch that data again, because it was populated from server-side (and hopefully you saved it on a store).

Continuing on the client-side now, when you visit different pages then Quasar checks if those pages have preFetch and if so, executes them. From what I see in your routes.js screenshot, the rendering page doesn't change when you go from /good/A to /good/B, because it's the same route. What you need is a watcher on the route parameter to trigger the fetch.
